app/models/user.rb

```diff
@@ -34,41 +34,41 @@ class User < ApplicationRecord
   has_many :given_lectures,
            class_name: "Lecture",
            foreign_key: "teacher_id",
-           inverse_of: :user
```

Comment:
Note that for this change (and all the others in the same commit) I opened a rails console and tested: `user.given lectures` would throw an exception with the `inverse_of: :user` option, whereas everything worked well with `inverse_of: :teacher`.